### DISCUSSION TITLE
LXQt Theme Desktop Wallpapers Changing with "Use wallpaper provided by theme" box unchecked, upon change of theme [SOLVED]

### DISCUSSION THREAD

**Discussion by edwardp3** [lxqt/lxqt#2368] [General]
LXQt 1.2.0 on Debian unstable/sid, amd64: 

On LXQt Appearance Configuration / LXQt Theme, the checkbox marked _Use wallpaper provided by theme_ is not checked by default. If the theme currently in use is changed and it also includes its own wallpaper, the desktop wallpaper is changing with the theme, with this box remaining unchecked. 

This occurs with the themes included with LXQt, as well as User themes separately installed to the `/.local/share/lxqt/themes` directory.

**Comment by M4he** [comment]
Can confirm. This is a weakness of the current implementation of lxqt-config and only triggers in a specific edge case:

The "Use wallpaper provided by theme" is ignored if the currently applied wallpaper matches exactly the wallpaper preset of the theme you are changing *from*. Even if you did explicitly set that wallpaper on your own, as soon as it matches with the preset of a theme that you are currently using and then switch to another, the wallpaper is replaced.

Example:

- `/usr/share/lxqt/themes/Clearlooks/wallpaper.cfg` points to `/usr/share/lxqt/wallpapers/waves-logo.png`

Let's say you manually configure `/usr/share/lxqt/wallpapers/waves-logo.png` as your wallpaper, then:

1. starting point: LXQt theme is set to 'Frost', wallpaper set to the aforementioned file
2. set LXQt theme to 'Kvantum', wallpaper stays because it didn't belong to 'Frost'
3. set LXQt theme to 'Clearlooks', wallpaper stays because it didn't belong to 'Kvantum' but it is theme same as would be applied by 'Clearlooks' if "Use wallpaper provided by theme" was enabled
4. set LXQt theme to something else, **now the wallpaper changes** because your wallpaper is an exact match of the preset of 'Clearlooks' which you just changed from, so lxqt-config *thinks* you were using the theme's wallpaper

An easy solution is to make a copy of the desired wallpaper from `/usr/share/lxqt/wallpapers/` with a different path/name. This way this edge case shouldn't trigger. If this behavior still bothers you, consider filing a bug report for lxqt-config.
The culprit should be [the `isWallpaperChanged` check here](https://github.com/lxqt/lxqt-config/blob/0698dc942f54b4afec730c2f9443f203945ed0dd/lxqt-config-appearance/lxqtthemeconfig.cpp#L154).

**Reply by edwardp3** [reply]
I just discovered that the checkbox works as expected, once the desktop wallpaper is first changed to something independent of a theme.
